awscli/customizations/s3/transferconfig.py

```diff
@@ -93,18 +96,44 @@ def _convert_human_readable_rates(self, runtime_config):
             value = runtime_config.get(attr)
             if value is not None and not isinstance(value, six.integer_types):
                 if value.endswith('B/s'):
-                    runtime_config[attr] = human_readable_to_int(value[:-2])
+                    runtime_config[attr] = self._human_readable_rate_to_int(
+                        value
+                    )
                 elif value.endswith('b/s'):
-                    bits_per_sec = human_readable_to_int(value[:-2])
+                    bits_per_sec = self._human_readable_rate_to_int(value)
                     bytes_per_sec = int(bits_per_sec / 8)
                     runtime_config[attr] = bytes_per_sec
+                elif self._INTEGER_STR_REGEX.match(value):
```

Comment:
Just curious why you went with a regex as opposed to a try/except. If a user passes in the value as `10.0` are we considering that as unsupported? Also it looks like the message only discusses rates. Do we want to update that to reflect this behavior?

Reply:
I went with the regex to avoid being entirely coupled to Python's `int()` interface. By restricting the values to be strings of digits 0-9, it ensures that we don't necessarily support all strings that Python supports for `int()` and we have more control over what we want to support. For example, I think you can specify underscores in string integer values:
```python
>>> int('1_000_000')
1000000
```
In the end, I don't have a strong preference about this. I could go either way. Thoughts?

> If a user passes in the value as 10.0 are we considering that as unsupported? 

Yeah that would not be supported. Right now all rates are in terms of bytes/sec and I don't think there is much value in supporting a fraction of a byte for these configurations.

> Also it looks like the message only discusses rates. Do we want to update that to reflect this behavior?

Good catch. I can update that.

Reply:
Following up on this. I guess we already sort of bounded to what Python accepts for `int()`. In the currently released logic, you can set the configuration rate value with underscores and it will properly interpret values like this: `8_000Mb/s`. I'm just going to update the logic to try to do an `int()` conversion instead of the regex to keep it all consistent. I don't think the current set of values that Python supports is an issue.